### PR TITLE
feat: support scaling in sql backend

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

It supports low-level operations but doesn't yet support automatic scaling (lacking the field for table parallelism).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/meta/src/controller/fragment.rs [hunk 1]
```diff
@@ -576,6 +576,26 @@ impl CatalogController {
...
 
+    pub async fn get_fragment_job_id(
+        &self,
+        fragment_ids: Vec<FragmentId>,
+    ) -> MetaResult<Vec<ObjectId>> {
+        let inner = self.inner.read().await;
+
+        let fragment_jobs: Vec<(FragmentId, ObjectId)> = Fragment::find()
+            .select_only()
+            .columns([fragment::Column::FragmentId, fragment::Column::JobId])
```
Comment (yezizp2012) [inline src/meta/src/controller/fragment.rs:587]: We can only query job_id here.

File: src/meta/src/controller/streaming_job.rs [hunk 2]
```diff
@@ -464,6 +492,22 @@ impl CatalogController {
             return Err(MetaError::permission_denied("table version is stale"));
         }
 
+        // let parallelism = match default_parallelism {
```
Comment (yezizp2012) [inline src/meta/src/controller/streaming_job.rs:495]: Can be removed.

File: src/meta/src/controller/streaming_job.rs [hunk 3]
```diff
@@ -831,4 +876,348 @@ impl CatalogController {
...
+                        .sorted()
+                        .collect_vec()
+                );
+
+                let actor_upstreams = ActorUpstreamActors(actor_upstreams);
+
+                let status = ActorStatus::from(PbActorState::try_from(state).unwrap());
+                let parallel_unit_id = parallel_unit.unwrap().id;
+
+                //                let status = actor_status.get_state().unwrap().into();
```
Comment (yezizp2012) [inline src/meta/src/controller/streaming_job.rs:1008]: Can remove it.

**Review by yezizp2012 — APPROVED** [review]
Thx for the job, generally LGTM. We need to rely on some tests to help us discover some corner cases, as well as take advantage of the benefits of the SQL backend to make necessary optimizations. We can merge it now and do it in later PRs.
